### ISSUE TITLE
taxii-server ignoring objects past v0.2.1

### ISSUE THREAD

**Issue by m0jtaba** [oasis-open/cti-taxii-server#47]
[Enterprise Attack JSON](https://raw.githubusercontent.com/mitre/cti/master/enterprise-attack/enterprise-attack.json)

```python 
import json
import socket
from time import sleep
import sys
from stix2 import TAXIICollectionSink
from taxii2client import Collection


def main():
    """
    ....
    """
    attack()


def attack():
    """
    Import attack into TAXII server
    """
    collection = Collection(
        "http://192.168.56.100/tde/collections/91a7b528-80eb-42ed-a74d-c6fbd5a26116/",
        user="guest",
        password="guest")
    # create TAXIICollectionSINK and push STIX content to it
    taxii_collection_sink = TAXIICollectionSink(collection)
    data = json.load(open('enterprise-attack.json'))
    taxii_collection_sink.add(data)
    print("Attack added to TAXII Server")


if __name__ == "__main__":
    main()
```

To recreate it grab the latest taxii server and run it. Then compare the the first object and it will be missing a lot of the fields.

Is this intended from version 0.2.1 upwards? If so how can I achieve the same with the latest version?

**Comment by m0jtaba** [comment]
Taxii server output objects[0]:

objects |  
-- | --
0 |  
created | "2017-12-14T16:46:06.044Z"
created_by_ref | "identity--c78cb6e5-0c4b-4611-8297-d1b8b55e40b5"
id | "attack-pattern--01df3350-ce05-4bdf-bdf8-0a919a66d4a8"
modified | "2018-10-31T13:45:13.024Z"
name | ".bash_profile and .bashrc"
object_marking_refs |  
0 | "marking-definition--fa42…-4e51-bc29-71d5b4802168"
type | "attack-pattern"


expected:

    "objects": [
        {
            "id": "attack-pattern--01df3350-ce05-4bdf-bdf8-0a919a66d4a8",
            "created_by_ref": "identity--c78cb6e5-0c4b-4611-8297-d1b8b55e40b5",
            "name": ".bash_profile and .bashrc",
            "description": "<code>~/.bash_profile</code> and <code>~/.bashrc</code> are executed in a user's context when a new shell opens or when a user logs in so that their environment is set correctly. <code>~/.bash_profile</code> is executed for login shells and <code>~/.bashrc</code> is executed for interactive non-login shells. This means that when a user logs in (via username and password) to the console (either locally or remotely via something like SSH), <code>~/.bash_profile</code> is executed before the initial command prompt is returned to the user. After that, every time a new shell is opened, <code>~/.bashrc</code> is executed. This allows users more fine grained control over when they want certain commands executed.\n\nMac's Terminal.app is a little different in that it runs a login shell by default each time a new terminal window is opened, thus calling <code>~/.bash_profile</code> each time instead of <code>~/.bashrc</code>.\n\nThese files are meant to be written to by the local user to configure their own environment; however, adversaries can also insert code into these files to gain persistence each time a user logs in or opens a new shell  (Citation: amnesia malware).",
            "external_references": [
                {
                    "external_id": "T1156",
                    "url": "https://attack.mitre.org/techniques/T1156",
                    "source_name": "mitre-attack"
                },
                {
                    "url": "https://researchcenter.paloaltonetworks.com/2017/04/unit42-new-iotlinux-malware-targets-dvrs-forms-botnet/",
                    "description": "Claud Xiao, Cong Zheng, Yanhui Jia. (2017, April 6). New IoT/Linux Malware Targets DVRs, Forms Botnet. Retrieved February 19, 2018.",
                    "source_name": "amnesia malware"
                }
            ],
            "object_marking_refs": [
                "marking-definition--fa42a846-8d90-4e51-bc29-71d5b4802168"
            ],
            "x_mitre_version": "1.0",
            "x_mitre_data_sources": [
                "File monitoring",
                "Process monitoring",
                "Process command-line parameters",
                "Process use of network"
            ],
            "x_mitre_detection": "While users may customize their <code>~/.bashrc</code> and <code>~/.bash_profile</code> files , there are only certain types of commands that typically appear in these files. Monitor for abnormal commands such as execution of unknown programs, opening network sockets, or reaching out across the network when user profiles are loaded during the login process.",
            "x_mitre_permissions_required": [
                "User",
                "Administrator"
            ],
            "x_mitre_platforms": [
                "Linux",
                "macOS"
            ],
            "type": "attack-pattern",
            "kill_chain_phases": [
                {
                    "phase_name": "persistence",
                    "kill_chain_name": "mitre-attack"
                }
            ],
            "modified": "2018-10-31T13:45:13.024Z",
            "created": "2017-12-14T16:46:06.044Z"
        },

**Comment by emmanvg** [comment]
@m0jtaba just for clarification:

1. The problem you show is that once you submit the file to the taxii server and pull it back it strips out all the custom properties?
2. What backend were you using? Memory or MongoDB (just to give me an idea on replicating this)

**Comment by m0jtaba** [comment]
@emmanvg

1. Yes that is correct.
2. MongoDB

**Comment by emmanvg** [comment]
@m0jtaba thanks! I can confirm the behavior and will look into it.

**Comment by emmanvg** [comment]
closed by #54. Changes will be present on next release.
